lib/notifications/notification.ex

```diff
@@ -72,6 +74,7 @@ defmodule Notifications.Notification do
       if db_record.id do
         notification_with_id = %__MODULE__{notification_without_id | id: db_record.id}
         log_creation(notification_with_id)
+        link_notification_to_users(notification_with_id)
```

Comment:
Might be good to put this inside of a transaction along with the `insert!` so that we don't get rows that are created but not linked to users.